Update Go and dependencies (#12)
tailscale/mkctr: README.md

```bash
mkctr \

>   --base="ghcr.io/tailscale/alpine-base:3.16" \

Reviewer: If we own this as well, should we just document `latest`?

Author: we don't push latest tag for this image

Reviewer 2: This is just an example, we can replace it with an upstream alpine image

`mkctr` auto discovers `GOOS`/`GOARCH` from the specified base image. If the base image supports multiple platforms, binaries are compiled for each platform as long as it's one of `linux/amd64`, `linux/386`, `linux/arm`, `linux/arm64`.

## Maturity

This is under active development. While Tailscale uses it, backwards compatability is not guaranteed, and some functionality is missing.

> ## Known issues

Author: I ran into this whilst trying to build a custom base for containerboot image and thought it'd be useful to add. Really, we should fix it, but might be low priority given that this is largely just for our images. Happy to not add this commit though if people don't like it

Reviewer 2: Maybe file a tracking issue instead?

Author: https://github.com/tailscale/mkctr/issues/13
seems like it is to do with the manifest structure, not with the type

Base images with media type `application/vnd.docker.distribution.manifest.v2+json` do not work with `mkctr`.

Known supported base image media types are `application/vnd.oci.image.index.v1+json`.

You can verify the media type of your image using [gcrane](https://github.com/google/go-containerregistry/tree/main/cmd/gcrane) and [jq](https://jqlang.github.io/jq/download/):

```
gcrane manifest <image> | jq .mediaType
```

> If you are building the base image with `docker buildx build ...` make sure you don't pass `--load` or `--output=type=docker` flags that would result in an image of docker media type.

Reviewer: ```suggestion
If you are building the base image with `docker buildx build ...` passing `--load` or `--output=type=docker` flags is known to result in an unsupported media type.
```

(just because it took me a few seconds to understand that "docker media type" refers to `application/vnd.docker.distribution.manifest.v2+json`)

Author: yeah I don't think 'docker media type' is really a thing
